### PR TITLE
Category mapper lower to llvm part2

### PR DESCRIPTION
**Pull request by etiotto**
This PR implements the support to lower the `CategoryMapper` operator from the Krnl dialect to the LLVM dialect. 
It also add LIT tests to verify that the operator is lowered to LLVM as expected.

### CODE REVIEW

**Comment by etiotto** [comment]
@jenkins-droid test this please

**Comment by etiotto** [comment]
@jenkins-droid test this please

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, not being a Krnl to LLVM expert, it might be good to have a review of someone more familiar with the pass.

File: test/mlir/krnl/krnl_category_mapper.mlir [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    } else {
+      krnl.store %c-1_i64, %0[%6#0, %6#1] : memref<2x2xi64>
+    }
+  }
+  return %0 : memref<2x2xi64>
+
+  // CHECK-DAG: llvm.func @strncmp(!llvm.ptr<i8>, !llvm.ptr<i8>, i64) -> i32
+  // CHECK-DAG: llvm.func @strlen(!llvm.ptr<i8>) -> i64
+  // CHECK-DAG: llvm.func @find_index_str(!llvm.ptr<i8>, !llvm.ptr<i32>, !llvm.ptr<i32>, i32) -> i32
+  // CHECK-DAG: llvm.mlir.global internal constant {{.*}}(dense<["cat", "dog", "cow"]> : tensor<3x!krnl.string>) : !llvm.array<3 x struct<(ptr<i8>, ptr<i8>, i64, array<1 x i64>, array<1 x i64>)>>
```
Comment (tungld) [inline test/mlir/krnl/krnl_category_mapper.mlir:83]: Just want to make sure that `!krnl.string` type can exit from MLIR and be lowered to LLVM properly. Have you checked that this generated code can be translated to LLVM and then to Bitcode?

**Pull request merged**